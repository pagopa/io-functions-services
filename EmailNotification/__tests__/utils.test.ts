import {
  contentToHtml,
  invalidateClickableLinks,
  messageReducedToHtml,
  removeLinks,
  truncateMarkdown
} from "../utils";
import * as E from "fp-ts/Either";
import {
  aCreatedMessageEventSenderMetadata,
  aMessageContent,
  anError
} from "../../__mocks__/mocks";

describe("contentToHtml", () => {
  test("GIVEN a simple content WHEN is converterd to html using the default processor THEN return an either containing the simple content", async () => {
    const simpleContent = "a simple content";
    const result = await contentToHtml()(simpleContent)();
    expect(result).toEqual(E.right(`<p>${simpleContent}</p>`));
  });

  test("GIVEN a multiline content WHEN is converterd to html using the default processor THEN return an either containing the content split in paragraph", async () => {
    const multilineContent = ["first line", "second line", "third line"];
    const result = await contentToHtml()(multilineContent.join("\n"))();
    expect(result).toEqual(
      E.right(`<p>${multilineContent.join("</p><p>")}</p>`)
    );
  });

  test("GIVEN any content WHEN is converterd to html using a not working processor THEN return an either containing an error", async () => {
    const mockProcessor = jest.fn().mockImplementationOnce(async () => {
      throw anError;
    });
    const result = await contentToHtml(mockProcessor)("")();
    expect(result).toEqual(E.left(anError));
  });
});

describe("messageReducedToHtml", () => {
  test("GIVEN a simple message WHEN is converted to an html using the default processor THEN return an either containing the email template with the simple content", async () => {
    const simpleMessage = {
      content: aMessageContent,
      senderMetadata: aCreatedMessageEventSenderMetadata
    };
    const result = await messageReducedToHtml()(simpleMessage)();
    expect(result).toMatchSnapshot();
  });

  test("GIVEN any message WHEN is converted to an html using a not working processor THEN return an either containing an error", async () => {
    const mockProcessor = jest.fn().mockImplementationOnce(async () => {
      throw anError;
    });
    const simpleMessage = {
      content: aMessageContent,
      senderMetadata: aCreatedMessageEventSenderMetadata
    };
    const result = await messageReducedToHtml(mockProcessor)(simpleMessage)();
    expect(result).toEqual(E.left(anError));
  });
});

describe("truncateMarkdown", () => {
  test("should add '...' at the end of the string if the plain text length is > 134", () => {
    expect(truncateMarkdown(aMessageContent.markdown).slice(-3)).toBe("...");
  });
  test("should not add '...' at the end of the string if the plain text length is <= 134", () => {
    expect(truncateMarkdown("This message is < than 134 chars")).not.toContain(
      "..."
    );
  });
});

describe("removeLinks", () => {
  test("should return the same string if no url is contained", () => {
    const text = "A simple text without any url";
    expect(removeLinks(text)).toBe(text);
  });

  test("should return the string without the url if a simple url is contained", () => {
    const simpleLink = "https://asimplelink.com/";
    const baseText = `A simple text`;
    expect(removeLinks(`${baseText} ${simpleLink}`)).toBe(
      `${baseText} [link rimosso]`
    );
  });

  test("should return the string without the url if more than one simple url are contained", () => {
    const simpleLink = "https://asimplelink.com/";
    const baseText = `A simple text`;
    expect(
      removeLinks(
        `${baseText} ${simpleLink} this is another link ${simpleLink}`
      )
    ).toBe(`${baseText} [link rimosso] this is another link [link rimosso]`);
  });

  test("should return the string without the url if an url with query params is contained", () => {
    const simpleLink = "https://asimplelink.com/?qp1=value";
    const baseText = `A simple text`;
    expect(removeLinks(`${baseText} ${simpleLink}`)).toBe(
      `${baseText} [link rimosso]`
    );
  });

  test("should return the string without the url if an url with query params and # is contained ", () => {
    const simpleLink = "https://asimplelink.com/?qp1=value#header";
    const baseText = `A simple text`;
    expect(removeLinks(`${baseText} ${simpleLink}`)).toBe(
      `${baseText} [link rimosso]`
    );
  });
});

describe("invalidateClickableLinks", () => {
  test("should return the same string if no period are provided", () => {
    expect(invalidateClickableLinks("a simple text with no period")).toBe(
      "a simple text with no period"
    );
  });

  test("should add a zero-width space before every '.' character", () => {
    expect(invalidateClickableLinks("a text.with 2 period.")).toBe(
      "a text.\u{200B}with 2 period.\u{200B}"
    );
  });
});
