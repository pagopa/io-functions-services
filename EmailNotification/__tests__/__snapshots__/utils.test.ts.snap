// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`messageToHtml GIVEN a simple message WHEN is converted to an html using the default processor THEN return an either contianing the email template with the simple content 1`] = `
Object {
  "_tag": "Right",
  "right": "
<!doctype html>
<html lang=\\"it\\" xmlns=\\"http://www.w3.org/1999/xhtml\\" xmlns:v=\\"urn:schemas-microsoft-com:vml\\" xmlns:o=\\"urn:schemas-microsoft-com:office:office\\">

<head>
  <title>
    {{title}}
  </title>
  <!--[if !mso]><!-->
  <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
  <!--<![endif]-->
  <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
  <style type=\\"text/css\\">
    #outlook a {
      padding: 0;
    }

    body {
      margin: 0;
      padding: 0;
      -webkit-text-size-adjust: 100%;
      -ms-text-size-adjust: 100%;
    }

    table,
    td {
      border-collapse: collapse;
      mso-table-lspace: 0pt;
      mso-table-rspace: 0pt;
    }

    img {
      border: 0;
      height: auto;
      line-height: 100%;
      outline: none;
      text-decoration: none;
      -ms-interpolation-mode: bicubic;
    }

    p {
      display: block;
      margin: 13px 0;
    }

  </style>
  <!--[if mso]>
    <noscript>
    <xml>
    <o:OfficeDocumentSettings>
      <o:AllowPNG/>
      <o:PixelsPerInch>96</o:PixelsPerInch>
    </o:OfficeDocumentSettings>
    </xml>
    </noscript>
    <![endif]-->
  <!--[if lte mso 11]>
    <style type=\\"text/css\\">
      .mj-outlook-group-fix { width:100% !important; }
    </style>
    <![endif]-->
  <!--[if !mso]><!-->
  <link href=\\"https://selfcare.pagopa.it/assets/font/selfhostedfonts.css\\" rel=\\"stylesheet\\" type=\\"text/css\\">
  <style type=\\"text/css\\">
    @import url(https://selfcare.pagopa.it/assets/font/selfhostedfonts.css);

  </style>
  <!--<![endif]-->
  <style type=\\"text/css\\">
    @media only screen and (min-width:480px) {
      .mj-column-per-100 {
        width: 100% !important;
        max-width: 100%;
      }

      .mj-column-per-50 {
        width: 50% !important;
        max-width: 50%;
      }
    }

  </style>
  <style media=\\"screen and (min-width:480px)\\">
    .moz-text-html .mj-column-per-100 {
      width: 100% !important;
      max-width: 100%;
    }

    .moz-text-html .mj-column-per-50 {
      width: 50% !important;
      max-width: 50%;
    }

  </style>
  <style type=\\"text/css\\">
    @media only screen and (max-width:480px) {
      table.mj-full-width-mobile {
        width: 100% !important;
      }

      td.mj-full-width-mobile {
        width: auto !important;
      }
    }

  </style>
  <style type=\\"text/css\\">
    :root {
      color-scheme: light dark;
      supported-color-schemes: light dark;
    }

    body,
    .mail-body {
      background-color: #FFFFFF;
    }

    .header {
      background-color: #FFFFFF;
      /* border-top: 8px solid #0073E6; */
    }

    .section {
      background-color: #FFFFFF;
    }

    .title div {
      /* Text Primary */
      color: #17324D !important;
    }

    .text div {
      /* Text Primary */
      color: #17324D !important;
    }

    .text-secondary,
    .text-secondary div {
      /* Text Secondary */
      color: #5C6F82 !important;
    }

    p {
      margin-top: 12px;
      margin-bottom: 12px;
    }

    .text-secondary a,
    .text a {
      color: #0073E6 !important;
    }

    .link {
      color: #0073E6 !important;
    }

    .divider p {
      border-color: #E3E7EB !important;
    }

    /* Message Preview Block */
    .message-preview {
      background-color: #FAFAFA;
    }

    @media (prefers-color-scheme: light) {

      .dark-img,
      [data-ogsc] .dark-img {
        display: none !important;
        mso-hide: all !important;
      }

      .light-img,
      [data-ogsc] .light-img {
        display: block;
        display: block !important;
      }
    }

    @media (min-width: 600px) {
      .mobile {
        display: none;
      }

      .desktop {
        display: block;
      }
    }

    @media (prefers-color-scheme: dark) {

      body,
      .mail-body {
        background-color: #323232;
      }

      .header {
        background-color: #323232;
        /* border-top: 8px solid #3da2ff; */
      }

      .section {
        background-color: #323232;
      }

      .message-preview {
        background-color: #424242;
      }

      .divider p {
        border-color: #757575 !important;
      }

      .title div {
        color: #ffffff !important;
      }

      .text div {
        color: #ffffff !important;
      }

      .text-secondary,
      .text-secondary div {
        /* Gray-200 */
        color: #E0E0E0 !important;
      }

      .text-secondary a,
      .text a {
        color: #3da2ff !important;
      }

      .link {
        color: #3da2ff !important;
      }

      .cta table td,
      .cta table td>a {
        background-color: #3da2ff !important;
      }

      /* IMAGES */
      /* Source: https://www.litmus.com/blog/the-ultimate-guide-to-dark-mode-for-email-marketers/ */
      .dark-img,
      [data-ogsc] .dark-img {
        display: block !important;
        overflow: visible !important;
        float: none !important;
        line-height: auto !important;
        margin-top: 0px !important;
        visibility: inherit !important;
      }

      .light-img,
      [data-ogsc] .light-img {
        display: none;
        display: none !important;
        mso-hide: all !important;
      }
    }

  </style>
  <meta http-equiv=\\"Content-Type\\" content=\\"text/html\\" charset=\\"UTF-8\\" />
  <meta name=\\"color-scheme\\" content=\\"light dark\\" />
  <meta name=\\"supported-color-schemes\\" content=\\"light dark\\" />
</head>

<body style=\\"word-spacing:normal;\\">
  <div class=\\"mail-body\\" style=\\"\\" lang=\\"it\\">
    <!--[if mso | IE]><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" role=\\"presentation\\" style=\\"width:600px;\\" width=\\"600\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->
    <div style=\\"margin:0px auto;max-width:600px;\\">
      <table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\">
        <tbody>
          <tr>
            <td style=\\"direction:ltr;font-size:0px;padding:20px 0;padding-left:20px;padding-right:20px;text-align:center;\\">
              <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><![endif]-->
              <!-- Entity Logo + App IO Logo -->
              <!--[if mso | IE]><tr><td class=\\"header-outlook\\" width=\\"600px\\" ><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"header-outlook\\" role=\\"presentation\\" style=\\"width:560px;\\" width=\\"560\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->
              <div class=\\"header\\" style=\\"margin:0px auto;max-width:560px;\\">
                <table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\">
                  <tbody>
                    <tr>
                      <td style=\\"direction:ltr;font-size:0px;padding:20px 0;padding-bottom:0px;padding-top:0px;text-align:center;\\">
                        <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" style=\\"width:560px;\\" ><![endif]-->
                        <div class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0;line-height:0;text-align:left;display:inline-block;width:100%;direction:ltr;\\">
                          <!--[if mso | IE]><table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" ><tr><td style=\\"vertical-align:middle;width:280px;\\" ><![endif]-->
                          <div class=\\"mj-column-per-50 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:middle;width:50%;\\">
                            <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:middle;\\" width=\\"100%\\">
                              <tbody>
                                <tr>
                                  <td align=\\"left\\" style=\\"font-size:0px;padding:10px 25px;padding-left:0px;word-break:break-word;\\">
                                    <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"border-collapse:collapse;border-spacing:0px;\\">
                                      <tbody>
                                        <tr>
                                          <td style=\\"width:64px;\\">
                                            <img alt=\\" Logo An Organization Name\\" height=\\"64\\" src=\\"https://assets.cdn.io.italia.it/logos/organizations/1234567890.png\\" style=\\"border:0;border-radius:8px;display:block;outline:none;text-decoration:none;height:64px;width:100%;font-size:13px;\\" width=\\"64\\" />
                                          </td>
                                        </tr>
                                      </tbody>
                                    </table>
                                  </td>
                                </tr>
                              </tbody>
                            </table>
                          </div>
                          <!--[if mso | IE]></td><td style=\\"vertical-align:middle;width:280px;\\" ><![endif]-->
                          <div class=\\"mj-column-per-50 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:middle;width:50%;\\">
                            <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:middle;\\" width=\\"100%\\">
                              <tbody>
                                <tr>
                                  <td align=\\"right\\" class=\\"light-img\\" style=\\"font-size:0px;padding:10px 25px;padding-right:0px;word-break:break-word;\\">
                                    <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"border-collapse:collapse;border-spacing:0px;\\">
                                      <tbody>
                                        <tr>
                                          <td style=\\"width:36px;\\">
                                            <img height=\\"auto\\" src=\\"https://raw.githubusercontent.com/pagopa/io-messages-email-templates/ICC-293--update-servicemessage-temaplte-with-placeholder/assets/logo_app_io.png\\" style=\\"border:0;display:block;outline:none;text-decoration:none;height:auto;width:100%;font-size:13px;\\" width=\\"36\\" />
                                          </td>
                                        </tr>
                                      </tbody>
                                    </table>
                                  </td>
                                </tr>
                                <div class=\\"dark-img\\" style=\\"mso-hide: all; display: none; padding: 10px 0px\\" align=\\"right\\">
                                  <img src=\\"https://raw.githubusercontent.com/pagopa/io-messages-email-templates/ICC-293--update-servicemessage-temaplte-with-placeholder/assets/logo_app_io_white.png\\" width=\\"36px\\" class=\\"dark-img\\" style=\\"display: none\\" />
                                </div>
                              </tbody>
                            </table>
                          </div>
                          <!--[if mso | IE]></td></tr></table><![endif]-->
                        </div>
                        <!--[if mso | IE]></td></tr></table><![endif]-->
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <!--[if mso | IE]></td></tr></table></td></tr><tr><td class=\\"section-outlook\\" width=\\"600px\\" ><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"section-outlook\\" role=\\"presentation\\" style=\\"width:560px;\\" width=\\"560\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->
              <div class=\\"section\\" style=\\"margin:0px auto;max-width:560px;\\">
                <table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\">
                  <tbody>
                    <tr>
                      <td style=\\"direction:ltr;font-size:0px;padding:20px 0;padding-bottom:0px;padding-top:0px;text-align:center;\\">
                        <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" style=\\"vertical-align:top;width:560px;\\" ><![endif]-->
                        <div class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\">
                          <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\">
                            <tbody>
                              <!-- Service Organization Name + Service Name --><!-- H6 Mobile (from MUI Italia) -->
                              <tr>
                                <td align=\\"left\\" class=\\"text\\" style=\\"font-size:0px;padding:10px 25px;padding-top:16px;padding-right:0px;padding-bottom:0px;padding-left:0px;word-break:break-word;\\">
                                  <div style=\\"font-family:Titillium Web, system-ui, sans-serif;font-size:18px;font-weight:600;line-height:24px;text-align:left;color:#17324D;\\">An Organization Name</div>
                                </td>
                              </tr>
                              <!-- Body 1 Desktop (from MUI Italia) -->
                              <tr>
                                <td align=\\"left\\" class=\\"text-secondary\\" style=\\"font-size:0px;padding:10px 25px;padding-top:0px;padding-right:0px;padding-left:0px;word-break:break-word;\\">
                                  <div style=\\"font-family:Titillium Web, system-ui, sans-serif;font-size:16px;font-weight:400;line-height:24px;text-align:left;color:#5C6F82;\\">aDepartmentName<br /> A_SERVICE_NAME</div>
                                </td>
                              </tr>
                              <!-- Title & Main Message -->
                              <tr>
                                <td align=\\"left\\" class=\\"title\\" style=\\"font-size:0px;padding:10px 25px;padding-top:24px;padding-right:0px;padding-bottom:0px;padding-left:0px;word-break:break-word;\\">
                                  <div style=\\"font-family:Titillium Web, system-ui, sans-serif;font-size:13px;font-weight:bold;line-height:1;text-align:left;color:#17324D;\\"><!-- H4 Desktop (from MUI Italia)-->
                                    <h1 style=\\"font-size: 32px; line-height: 40px; margin: 0\\"> testtesttesttesttesttesttesttesttesttest </h1>
                                  </div>
                                </td>
                              </tr>
                              <tr>
                                <td align=\\"left\\" class=\\"text\\" style=\\"font-size:0px;padding:10px 25px;padding-top:4px;padding-right:0px;padding-left:0px;word-break:break-word;\\">
                                  <div style=\\"font-family:Titillium Web, system-ui, sans-serif;font-size:16px;font-weight:regular;line-height:24px;text-align:left;color:#17324D;\\"><!-- Body 1 Desktop (from MUI Italia)--> <p>testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest</p></div>
                                </td>
                              </tr>
                            </tbody>
                          </table>
                        </div>
                        <!--[if mso | IE]></td></tr></table><![endif]-->
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <!--[if mso | IE]></td></tr></table></td></tr><![endif]-->
              <!-- Divider -->
              <!--[if mso | IE]><tr><td class=\\"section-outlook\\" width=\\"600px\\" ><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"section-outlook\\" role=\\"presentation\\" style=\\"width:560px;\\" width=\\"560\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->
              <div class=\\"section\\" style=\\"margin:0px auto;max-width:560px;\\">
                <table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\">
                  <tbody>
                    <tr>
                      <td style=\\"direction:ltr;font-size:0px;padding:20px 0;padding-bottom:0px;padding-top:0px;text-align:center;\\">
                        <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" style=\\"vertical-align:top;width:560px;\\" ><![endif]-->
                        <div class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\">
                          <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\">
                            <tbody>
                              <tr>
                                <td align=\\"center\\" class=\\"divider\\" style=\\"font-size:0px;padding:0px;padding-top:16px;padding-bottom:16px;word-break:break-word;\\">
                                  <p style=\\"border-top:solid 1px #E3E7EB;font-size:1px;margin:0px auto;width:100%;\\">
                                  </p>
                                  <!--[if mso | IE]><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"border-top:solid 1px #E3E7EB;font-size:1px;margin:0px auto;width:560px;\\" role=\\"presentation\\" width=\\"560px\\" ><tr><td style=\\"height:0;line-height:0;\\"> &nbsp;
</td></tr></table><![endif]-->
                                </td>
                              </tr>
                            </tbody>
                          </table>
                        </div>
                        <!--[if mso | IE]></td></tr></table><![endif]-->
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <!--[if mso | IE]></td></tr></table></td></tr><![endif]-->
              <!-- Footer -->
              <!--[if mso | IE]><tr><td class=\\"section-outlook\\" width=\\"600px\\" ><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"section-outlook\\" role=\\"presentation\\" style=\\"width:560px;\\" width=\\"560\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->
              <div class=\\"section\\" style=\\"margin:0px auto;max-width:560px;\\">
                <table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\">
                  <tbody>
                    <tr>
                      <td style=\\"direction:ltr;font-size:0px;padding:20px 0;padding-top:0px;text-align:center;\\">
                        <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" style=\\"vertical-align:top;width:560px;\\" ><![endif]-->
                        <div class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\">
                          <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\">
                            <tbody>
                              <tr>
                                <td align=\\"left\\" class=\\"text-secondary\\" style=\\"font-size:0px;padding:0px;word-break:break-word;\\">
                                  <div style=\\"font-family:Titillium Web, system-ui, sans-serif;font-size:14px;font-weight:regular;line-height:20px;text-align:left;color:#5C6F82;\\">
                                    <p>
                                      <strong> Non rispondere a questo indirizzo email: è utilizzato solo per l'invio di questa comunicazione e, non essendo monitorato, non riceveresti risposta. </strong>
                                    </p>
                                    <p> Ricevi questa comunicazione perché hai abilitato l’inoltro dei messaggi via email per tutti i servizi di IO. </p>
                                    <p> Se non vuoi più che i messaggi che ricevi su IO vengano inoltrati al tuo indirizzo email, vai alla sezione Profilo > Preferenze > Inoltro dei messaggi via email e seleziona l'opzione “Disabilita per tutti i servizi”. </p>
                                  </div>
                                </td>
                              </tr>
                            </tbody>
                          </table>
                        </div>
                        <!--[if mso | IE]></td></tr></table><![endif]-->
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <!--[if mso | IE]></td></tr></table></td></tr></table><![endif]-->
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <!--[if mso | IE]></td></tr></table><![endif]-->
  </div>
</body>

</html>
",
}
`;
